unyt/array.py

```diff
@@ -265,7 +266,12 @@ def _coerce_iterable_units(input_object, registry=None):
                 return unyt_array(np.array(ret), ff, registry=registry)
             # This will create a copy of the data in the iterable.
             return unyt_array(np.array(input_object), ff, registry=registry)
```

Comment:
(to go with my previous suggestion)
```suggestion
            else:
                ret = unyt_array(np.array(input_object), ff, registry=registry)
```